roles/dns_client/vars/main.yml

```diff
@@ -0,0 +1,5 @@
+---
+network_file: /etc/sysconfig/network-scripts/ifcfg-eth0
+dns_domain: "{{ lookup('env', 'CLOUD_DOMAIN')|default('example.com', true) }}"
+        # FIXME getting the DNS server IP address
+dns_server: "{{ hostvars['dns.example.com'].openshift.common.ip }}"
```

Comment:
you should have a variable for the dns server ip, that is required by the dns_client module.  you can then set it in the playbook using groups.dns.openshift.common.ip